server/main_server/src/rodsServer.cpp

```diff
@@ -398,10 +397,11 @@ namespace
         return _default;
     } // get_advanced_setting
 
-    auto get_cache_clearer_sleep_time(const char* _keyword, int _default) -> decltype(_default)
+    auto get_cache_clearer_sleep_time(const char* _keyword, int _default) -> int
     {
         try {
-            const auto adv_settings = irods::server_properties::instance().map().get_json().at(irods::KW_CFG_ADVANCED_SETTINGS);
+            const auto config_handle = irods::server_properties::instance().map();
+            const auto adv_settings = config_handle.get_json().at(irods::KW_CFG_ADVANCED_SETTINGS);
```

Comment:
`const auto&`?

I just realized that L384/385 also needs to be a reference.

Reply:
Yep, done